### DISCUSSION TITLE
现在debug_console只能获取到lua服务的信息，c服务应该也定义一些供采集信息的接口。

### DISCUSSION THREAD

**Discussion by yuanfengyun** [cloudwu/skynet#1591] [Ideas]
现在debug_console只能获取到lua服务的信息，c服务应该也定义一些供采集信息的接口。

**Comment by cloudwu** [comment]
debug console 可以对任意服务发送消息，所以，只要你的 C 服务能相应定义好的消息就可以。你可以在 C 服务嵌入一个 lua vm 去解析 lua 消息。

**Reply by yuanfengyun** [reply]
现有的几个C服务没有提供这些接口。

**Reply by cloudwu** [reply]
现在已经没有 C 服务了，logger 服务也可以自定义。

**Comment by yuanfengyun** [comment]
还有几个内置的C服务。比如habor

**Reply by cloudwu** [reply]
目前代码库中只有三个 C 服务：

harbor 模式已经不推荐使用，建议用 cluster 代替。而且 harbor 的主要业务逻辑在 cmaster 和 cslave 两个 lua 服务中。

gate 纯粹为兼容存在，有完全对应的 gate lua 服务。

logger 是可选的，建议用自定义 logger 满足你的项目需要。

**Comment by yuanfengyun** [comment]
现在我们的项目中用了大量skynet.send(全局名字)，这样的用法，我想通过debug_console看看habor的转发会不会有什么问题。habor的CPU和消息队列的情况。发现做不到

**Comment by yuanfengyun** [comment]
如果是定义一个简单的文本回调，C服务就能很方便地影响debug_console的命令，现在这种太麻烦了。

**Comment by cloudwu** [comment]
不要用 global name ， 用 . 开头的 local name 。
更推荐的方法是用 service 模块：https://github.com/cloudwu/skynet/blob/master/lualib/skynet/service.lua

**Comment by cloudwu** [comment]
skynet 的发展趋势是尽量去掉所有 C 服务，全部迁移到 lua 服务中。所有 C 服务相关的代码只减不增。
